memory/memory.go

```diff
@@ -3,21 +3,29 @@ package queue
 import (
 	"sync"
 	"time"
+
+	"gopkg.in/src-d/go-queue.v0"
 )
 
+func init() {
+	queue.Register("memory", func(uri string) (queue.Broker, error) {
+		return NewMemoryBroker(), nil
+	})
+}
+
 type memoryBroker struct {
-	queues map[string]Queue
+	queues map[string]queue.Queue
 }
 
-// Creates a new Broker for an in-memory queue.
-func NewMemoryBroker() Broker {
-	return &memoryBroker{make(map[string]Queue)}
+// NewMemoryBroker creates a new Broker for an in-memory queue.
+func NewMemoryBroker() queue.Broker {
```

Comment:
Given that it is on its own package, we might want to call this just `NewBroker` or `New`. Same with the AMQP one.